libandroid-navigation/src/main/java/com/mapbox/services/android/navigation/v5/internal/navigation/NavigationEngineFactory.kt

```diff
@@ -0,0 +1,46 @@
+package com.mapbox.services.android.navigation.v5.internal.navigation
+
+import com.mapbox.services.android.navigation.v5.navigation.camera.Camera
+import com.mapbox.services.android.navigation.v5.navigation.camera.SimpleCamera
+import com.mapbox.services.android.navigation.v5.offroute.OffRoute
+import com.mapbox.services.android.navigation.v5.offroute.OffRouteDetector
+import com.mapbox.services.android.navigation.v5.route.FasterRoute
+import com.mapbox.services.android.navigation.v5.route.FasterRouteDetector
+import com.mapbox.services.android.navigation.v5.snap.Snap
+import com.mapbox.services.android.navigation.v5.snap.SnapToRoute
+
+class NavigationEngineFactory {
+
+    private var offRouteEngine: OffRoute = OffRouteDetector()
+    private var fasterRouteEngine: FasterRoute = FasterRouteDetector()
+    private var snapEngine: Snap = SnapToRoute()
+    private var cameraEngine: Camera = SimpleCamera()
+
+    fun retrieveOffRouteEngine(): OffRoute? {
```

Comment:
Could this be nullable?

Reply:
We can make the return type non-nullable since it can never be null

Reply:
👍